### PR TITLE
Better support for ModelAttribute

### PR DESCRIPTION
**Pull request by AnWeber**
fix for #355 
ModelAttribute is an optional annotation. By default, any argument that is not a simple value type (as determined by BeanUtils#isSimpleProperty) and is not resolved by any other argument resolver is treated as if it were annotated with @ModelAttribute.
This pull request fixes also section header. If operation is annotated with RequestMethod GET, section header is "query parameter".

### CODE REVIEW

File: spring-auto-restdocs-core/src/main/java/capital/scalable/restdocs/util/HandlerMethodUtil.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+ */
+package capital.scalable.restdocs.util;
+
+import static org.apache.commons.lang3.StringUtils.isNotBlank;
+import static org.springframework.util.StringUtils.arrayToDelimitedString;
+
+import org.springframework.core.MethodParameter;
+import org.springframework.web.method.HandlerMethod;
+
+import capital.scalable.restdocs.request.RequestParametersSnippet;
```
[the remark this answers is not in the document]
Comment (AnWeber) [reply spring-auto-restdocs-core/src/main/java/capital/scalable/restdocs/util/HandlerMethodUtil.java]: removed unused imports

File: spring-auto-restdocs-core/src/main/java/capital/scalable/restdocs/util/HandlerMethodUtil.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package capital.scalable.restdocs.util;
+
+import static org.apache.commons.lang3.StringUtils.isNotBlank;
+import static org.springframework.util.StringUtils.arrayToDelimitedString;
```
[the remark this answers is not in the document]
Comment (AnWeber) [reply spring-auto-restdocs-core/src/main/java/capital/scalable/restdocs/util/HandlerMethodUtil.java]: removed unused imports

File: spring-auto-restdocs-core/src/main/java/capital/scalable/restdocs/payload/AbstractJacksonFieldSnippet.java [hunk 3]
```diff
@@ -146,4 +146,17 @@ public String getFileName() {
...
+    @Override
+    protected String[] getTranslationKeys() {
+        return new String[]{
+            "th-parameter",
+            "th-type",
+            "th-optional",
+            "th-description",
+            "pagination-request-adoc",
+            "pagination-request-md",
+            "no-params"
```
[the remark this answers is not in the document]
Comment (AnWeber) [reply spring-auto-restdocs-core/src/main/java/capital/scalable/restdocs/payload/AbstractJacksonFieldSnippet.java]: There is no valid reason. I moved method to JacksonModelAttributeSnippet.

File: spring-auto-restdocs-core/src/main/java/capital/scalable/restdocs/payload/JacksonModelAttributeSnippet.java [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+    private boolean isModelAttribute(MethodParameter param) {
+        return param.getParameterAnnotation(ModelAttribute.class) != null;
+    }
+
+    private boolean hasNoHandlerMethodArgumentResolver(MethodParameter param) {
+        return !BeanUtils.isSimpleProperty(param.getParameterType())
+            &&  this.handlerMethodArgumentResolvers != null
+            && this.handlerMethodArgumentResolvers
+            .stream()
+            .allMatch(obj -> !obj.supportsParameter(param));
```
[the remark this answers is not in the document]
Comment (AnWeber) [reply spring-auto-restdocs-core/src/main/java/capital/scalable/restdocs/payload/JacksonModelAttributeSnippet.java]: thanks for your hint. I changed code to your solution.

**Comment by m-titov** [comment]
It seems this logic is incorrect. For example, @RequestHeader shouldn't be treat as request parameter

**Pull request merged**